File: CS50x/sentimental-readability/readability.py

from cs50 import get_string

wordCount = 0
letterCount = 0
sentencesCount = 0


def main():
    text = get_string("Give it to me baby: ")

    letterCount = countLetters(text)
    # print(f"Number of Letters: {letterCount}")

    wordCount = countWords(text)
    # print(f"Number of Words: {wordCount}")

    sentencesCount = countSentences(text)
    # print(f"Number of Sentences: {sentencesCount}")

    level = round(formula(letterCount, wordCount, sentencesCount))
    if level < 1:
        print("Before Grade 1")
    elif level > 16:
        print("Grade 16+")
    else:
        print(f"Grade {level}")


def countLetters(text):
    count = 0
    length = len(text)
    for i in range(0, length, 1):
        tmp = text[i]
        if tmp.isalpha():
            count += 1
    return count


def countWords(text):
    tmp = 0
    count = 1
    length = len(text)
    for i in range(0, length, 1):
        tmp = text[i]
        if tmp.isspace():
            count += 1
    return count


def countSentences(text):
    tmp = 0
    count = 0
    length = len(text)
    for i in range(0, length, 1):
        tmp = text[i]
        if (tmp == "." or tmp == "?" or tmp == "!"):
            count += 1
    return count

# Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8, where L is the average number of letters per 100 words in the text,
# and S is the average number of sentences per 100 words in the text
# L = (letters / (float) words) * 100; S = (sentences / (float) words) * 100;


def formula(letters, words, sentences):
    L = (letters / words) * 100
    S = (sentences / words) * 100

    lvl = 0.0588 * L - 0.296 * S - 15.8
    return lvl


if __name__ == "__main__":
    main()

File: CS50x/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int letters = 0;
int words = 0;
int sentences = 0;

int main(void)
{
    string text = get_string("Your text: ");

    letters = count_letters(text);
    //printf("Total letters: %i\n", letters);

    words = count_words(text);
    //printf("Total word: %i\n", words);

    sentences = count_sentences(text);
    //printf("Total sentences: %i\n", sentences);

    float L = (letters / (float) words) * 100;
    float S = (sentences / (float) words) * 100;

    float subindex = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(subindex);

    //printf("Index is : %i\n", index);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int total = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            total++;
            //printf("Total letters: %i\n", letter);
        }
    }
    return (total);
}

int count_words(string text)
{
    int total = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            total++;
        }
    }
    return (total);
}

int count_sentences(string text)
{
    int total = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (((text[i]) == 46) || ((text[i]) == 33) || ((text[i]) == 63))
        {
            total++;
        }
    }
    return (total);
}

